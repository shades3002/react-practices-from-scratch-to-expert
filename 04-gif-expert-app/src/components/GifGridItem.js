import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ id, title, url}) => {

    return (
        <div className="card animate__animated animate__bounce">
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}

GifGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}   