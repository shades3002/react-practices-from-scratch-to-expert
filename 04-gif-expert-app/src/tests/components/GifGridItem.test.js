import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Test in <GitGridItem />", () => {

  test("should display the component correctly", () => {

    const title = 'Goku';
    const url   = 'https://localhost/goku.jpg';

    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );
    expect(wrapper).toMatchSnapshot();

  });

});
