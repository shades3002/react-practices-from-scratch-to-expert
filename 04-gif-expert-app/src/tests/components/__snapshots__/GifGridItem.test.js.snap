// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GitGridItem /> should display the component correctly 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Goku"
    src="https://localhost/goku.jpg"
  />
  <p>
    Goku
  </p>
</div>
`;
